### PR TITLE
Initial Azure TF commit with AKS and ACR modules

### PR DESCRIPTION
**Pull request by pvijay-enquizit**

### CODE REVIEW

File: terraform/azure/modules/aks/aks-sub-net.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+//existing vnet
+data "azurerm_virtual_network" "existing-vnet" {
+  name                = "csels-nbs-dev-low-modern-vnet" #Need to create a var
+  resource_group_name = var.modern_resource_group_name
+}
+
+resource "azurerm_subnet" "aks-default" {
+  name                 = "csels-nbs-dev-low-modern-vnet-aks-sg" #Need to create a var
```
Comment (szamfir) [inline terraform/azure/modules/aks/aks-sub-net.tf:9]: name to switch to resource_prefix

File: terraform/azure/modules/aks/aks.tf [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    display_name =  var.azuread_service_principal_display_name
+}
+
+data "azurerm_resource_group" "rg" {
+    name     = var.modern_resource_group_name 
+}
+
+resource "azurerm_kubernetes_cluster" "aks" {
+  location            = data.azurerm_resource_group.rg.location
+  name                = var.k8_cluster_name
```
Comment (szamfir) [inline terraform/azure/modules/aks/aks.tf:28]: name to switch to resource_prefix

File: terraform/azure/modules/aks/log-analytics-workspace.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Create Log Analytics Workspace
+resource "azurerm_log_analytics_workspace" "insights" {
+  name                = "logs-${random_pet.aksrandom.id}"
```
Comment (szamfir) [inline terraform/azure/modules/aks/log-analytics-workspace.tf:3]: name to switch to resource_prefix

File: terraform/azure/modules/aks/aks.tf [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+data "azuread_service_principal" "akssp"{
+    display_name =  var.azuread_service_principal_display_name
+}
+
+data "azurerm_resource_group" "rg" {
+    name     = var.modern_resource_group_name 
+}
+
+resource "azurerm_kubernetes_cluster" "aks" {
```
Comment (szamfir) [inline terraform/azure/modules/aks/aks.tf:26]: I thought we were going to use the terraform module for the cluster creation? This is so we do not have to manage all the individual pieces. Just like we do with eks.
https://registry.terraform.io/modules/Azure/aks/azurerm/latest?tab=resources

File: terraform/azure/modules/aks/ssh.tf [hunk 5]
```diff
@@ -0,0 +1,24 @@
+resource "random_pet" "ssh_key_name" {
```
Comment (szamfir) [inline terraform/azure/modules/aks/ssh.tf:1]: Will this be recreated on every deployment? I believe that is what happens with random numbers. If not, then good to go

**Review by szamfir — CHANGES_REQUESTED** [review]
Names for created resource should follow our resource_prefix convention. See eks_nbs module. We do not need to provide a name override this time around since the Azure environments will be fresh.

I have marked where the names need to be changed with line comments.

Additionally, had a rather larger comment on how we are creating the AKS cluster.

File: terraform/azure/modules/aks/aks.tf (hunk 4 above)
Comment (pvijay-enquizit) [reply terraform/azure/modules/aks/aks.tf:26]: This is part of Azurerm.

File: terraform/azure/modules/aks/ssh.tf (hunk 5 above)
Comment (pvijay-enquizit) [reply terraform/azure/modules/aks/ssh.tf:1]: I dont see it getting created every time I deploy this.

File: terraform/azure/modules/aks/aks-sub-net.tf (hunk 1 above)
Comment (pvijay-enquizit) [reply terraform/azure/modules/aks/aks-sub-net.tf]: addressed

File: terraform/azure/modules/aks/aks.tf (hunk 2 above)
Comment (pvijay-enquizit) [reply terraform/azure/modules/aks/aks.tf]: addressed

File: terraform/azure/modules/aks/log-analytics-workspace.tf (hunk 3 above)
Comment (pvijay-enquizit) [reply terraform/azure/modules/aks/log-analytics-workspace.tf]: Fixed

**Review by szamfir — APPROVED** [review]
LGTM

**Pull request merged**